core/src/main/java/io/grpc/internal/TransportSet.java

```diff
@@ -280,16 +280,26 @@ public ClientTransport get() {
       }
     }
 
-    reconnectTask = scheduledExecutor.schedule(
-        new LogExceptionRunnable(new EndOfCurrentBackoff()), delayMillis, TimeUnit.MILLISECONDS);
     return new Runnable() {
       @Override
       public void run() {
         // This must be run outside of lock. The TransportSet lock is a channel level lock.
         // startBackoff() will acquire the delayed transport lock, which is a transport level
         // lock. Our lock ordering mandates transport lock > channel lock.  Otherwise a deadlock
         // could happen (https://github.com/grpc/grpc-java/issues/2152).
+        if (delayedTransport.isInBackoffPeriod()) {
```

Comment:
this code shouldn't be there, was used to compare travis mac results, actually was removed in my draft